controllers/component/component_adapter.go

```diff
@@ -105,12 +106,17 @@ func (a *Adapter) EnsureComponentIsCleanedUp() (controller.OperationResult, erro
 			return controller.RequeueWithError(err)
 		}
 	}
-	err = h.RemoveFinalizerFromComponent(a.client, a.logger, a.context, a.component, h.ComponentFinalizer)
-	if err != nil {
-		return controller.RequeueWithError(fmt.Errorf("failed to remove the finalizer: %w", err))
+	a.component, err = a.loader.GetComponent(a.client, a.context, a.component.Name, a.component.Namespace)
+	if errors.IsNotFound(err) {
+		err = h.RemoveFinalizerFromComponent(a.client, a.logger, a.context, a.component, h.ComponentFinalizer)
```

Comment:
what's the value of `a.component` when there is not found error? Nil or empty?

Reply:
Updated the logic, let me know if its correct now.